lib/t_web/views/match_view.ex

```diff
@@ -38,6 +38,74 @@ defmodule TWeb.MatchView do
     |> maybe_put("voicemail", voicemail)
   end
 
+  def render("interaction.json", %{interaction: interaction}) do
+    %Interaction{data: %{"type" => type}} = interaction
+    render_interaction(type, interaction)
+  end
+
+  defp render_interaction("slots_offer" = type, interaction) do
+    %Interaction{id: id, to_user_id: picker, data: %{"slots" => slots}} = interaction
+
+    %{
+      "id" => id,
+      "type" => type,
+      "slots" => slots,
+      "inserted_at" => datetime(id),
+      "picker" => picker
+    }
+  end
+
+  defp render_interaction("slot_accept" = type, interaction) do
+    %Interaction{id: id, data: %{"slot" => slot}} = interaction
+    %{"id" => id, "type" => type, "selected_slot" => slot, "accepted_at" => datetime(id)}
+  end
+
+  defp render_interaction("slot_cancel" = type, interaction) do
+    %Interaction{id: id, from_user_id: by_user_id} = interaction
+    %{"id" => id, "type" => type, "by_user_id" => by_user_id, "cancelled_at" => datetime(id)}
+  end
+
+  defp render_interaction("contact_offer" = type, interaction) do
+    %Interaction{id: id, data: %{"contacts" => contacts}, to_user_id: picker} = interaction
+
+    %{
+      "id" => id,
+      "type" => type,
+      "contacts" => contacts,
+      "picker" => picker,
+      "inserted_at" => datetime(id)
+    }
+  end
+
+  defp render_interaction("contact_cancel" = type, interaction) do
+    %Interaction{id: id, data: _data, from_user_id: by_user_id} = interaction
+    %{"id" => id, "type" => type, "by" => by_user_id, "at" => datetime(id)}
+    # |> maybe_put("cancels", data["id"])
+  end
+
+  defp render_interaction("voicemail" = type, interaction) do
+    %Interaction{id: id, from_user_id: caller, data: %{"s3" => s3_key}} = interaction
+
+    %{
+      "id" => id,
+      "type" => type,
+      "caller" => caller,
+      "s3_key" => s3_key,
+      "url" => Calls.voicemail_url(s3_key),
+      "inserted_at" => datetime(id)
+    }
+  end
+
+  defp render_interaction("call_attempt" = type, interaction) do
+    %Interaction{id: id, from_user_id: caller} = interaction
+    %{"id" => id, "type" => type, "caller" => caller, "attempted_at" => datetime(id)}
+  end
+
+  defp render_interaction("call_accepted", interaction) do
```

Comment:
there also needs to be a `call_ended` interaction with duration

Reply:
Duration should be calculated on the client since backend doesn't know in what format / units the client wants it.